src/rust/ide/src/controller/graph.rs

```diff
@@ -118,29 +107,16 @@ impl Handle {
     /// Creates a new controller. Does not check if id is valid.
     ///
     /// Requires global executor to spawn the events relay task.
-    pub fn new_unchecked(module:controller::module::Handle, id:Id) -> Handle {
-        let graphs_notifications = module.subscribe_graph_notifications();
-        let publisher = default();
+    pub fn new_unchecked(module:Rc<model::Module>, parser:Parser, id:Id) -> Handle {
         let logger    = Logger::new(format!("Graph Controller {}", id));
-        let ret       = Handle {module,id,publisher,logger};
-        let weak      = Rc::downgrade(&ret.publisher);
-        let relay_notifications = process_stream_with_handle(graphs_notifications,weak,
-            |notification,this| {
-                match notification {
-                    controller::notification::Graphs::Invalidate =>
-                    this.borrow_mut().publish(controller::notification::Graph::Invalidate),
-            }
-        });
-        executor::global::spawn(relay_notifications);
-        ret
+        Handle {module,parser,id,logger}
     }
 
-    /// Creates a new graph controller. Given ID should uniquely identify a definition in the
     /// module. Fails if ID cannot be resolved.
     ///
     /// Requires global executor to spawn the events relay task.
-    pub fn new(module:controller::module::Handle, id:Id) -> FallibleResult<Handle> {
-        let ret = Self::new_unchecked(module,id);
+    pub fn new(module:Rc<model::Module>, parser:Parser, id:Id) -> FallibleResult<Handle> {
```

Comment:
Is it really inconsistent? 
`model::Model` is a data structure, essentially a structure with data. Here we want to use some shared data, hence the Rc
`parser::Parser` is not a structure we want for keeping some data. It is just "essential data required for parsing things". It could have no Rcs inside as well and we would be happy with that.